utils_cv/detection/model.py

```diff
@@ -152,18 +257,43 @@ def _calculate_ap(e: CocoEvaluator) -> float:
     return np.mean(np.mean(coco_eval[precision_settings]))
 
 
+def _get_num_classes(dataset: Optional[DetectionDataset]) -> int:
+    return (
+        len(dataset.labels) + 1
+        if dataset and "labels" in dataset.__dict__
+        else len(coco_labels())
+    )
+
+
 class DetectionLearner:
     """ Detection Learner for Object Detection"""
 
-    def __init__(self, dataset: Dataset, model: nn.Module = None):
+    def __init__(
+        self,
+        dataset: DetectionDataset = None,
+        model: nn.Module = None,
+        device: torch.device = None,
+    ):
         """ Initialize leaner object. """
-        self.device = torch_device()
-        self.model = model
+        self.device = device
+        if self.device is None:
+            self.device = torch_device()
+
         self.dataset = dataset
 
+        if dataset and "labels" in dataset.__dict__:
+            self.labels = ["__background__"] + dataset.labels
+        else:
```

Comment:
not sure if we should default to coco labels if not specified

Reply:
Done.  Saw you added `add_labels()` method.